pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/ExcessiveLengthRule.java

```diff
@@ -15,7 +16,11 @@
  *
  * <p>To implement an ExcessiveLength rule, you pass in the Class of node you want
  * to check, and this does the rest for you.</p>
+ *
+ * @deprecated This base class is not needed anymore and will be removed with PMD 7.
  */
+@Deprecated
+@InternalApi
```

Comment:
we don't need to flag it as `@InternalApi` if it's intended for removal. We only use that for things we want to move into `internal` packages.

Reply:
:+1: